robot/impl/resource_manager.go

```diff
@@ -369,6 +369,30 @@ func (manager *resourceManager) ResourceNames() []resource.Name {
 	return names
 }
 
+// ResourceStatuses returns the names of all resources in the manager.
+func (manager *resourceManager) ResourceStatuses() []resource.Status {
+	result := []resource.Status{}
+	for _, name := range manager.resources.Names() {
+		if name.API == client.RemoteAPI {
+			continue
+		}
+		if name.API.Type.Namespace == resource.APINamespaceRDKInternal {
+			continue
+		}
```

Comment:
this is where I make the slight change to the `MachineStatus` API that I note in the PR description:

> MachineStatus changes slightly - it will no longer include RDK internal resources and RemoteAPIs (e.g. the resource that represents the remote itself). This mirrors the behavior of ResourceNames.